includes/rest/class-outbox-controller.php

```diff
@@ -0,0 +1,296 @@
+						'type'              => 'string',
+						'validate_callback' => array( $this, 'validate_user_id' ),
+					),
+				),
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( 'Activitypub\Rest\Server', 'verify_signature' ),
+					'args'                => array(
+						'page'     => array(
+							'description' => 'Current page of the collection.',
+							'type'        => 'integer',
+							'default'     => 1,
+							'minimum'     => 1,
+						),
+						'per_page' => array(
+							'description' => 'Maximum number of items to be returned in result set.',
+							'type'        => 'integer',
+							'default'     => 10,
+							'minimum'     => 1,
+							'maximum'     => 100,
+						),
+					),
+				),
+				'schema' => array( $this, 'get_collection_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Validates the user_id parameter.
+	 *
+	 * @param mixed $user_id The user_id parameter.
+	 * @return bool|\WP_Error True if the user_id is valid, WP_Error otherwise.
+	 */
+	public function validate_user_id( $user_id ) {
+		$user = Actors::get_by_various( $user_id );
+		if ( \is_wp_error( $user ) ) {
+			return $user;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Retrieves a collection of outbox items.
+	 *
+	 * @param \WP_REST_Request $request Full details about the request.
+	 * @return \WP_REST_Response|\WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		$user_id = $request->get_param( 'user_id' );
+		$page    = $request->get_param( 'page' );
+		$user    = Actors::get_by_various( $user_id );
+
+		/**
+		 * Action triggered prior to the ActivityPub profile being created and sent to the client.
+		 *
+		 * @param \WP_REST_Request $request The request object.
+		 */
+		\do_action( 'activitypub_rest_outbox_pre', $request );
+
+		/**
+		 * Filters the list of activity types to include in the outbox.
+		 *
+		 * @param string[] $activity_types The list of activity types.
+		 */
+		$activity_types = apply_filters( 'rest_activitypub_outbox_activity_types', array( 'Announce', 'Create', 'Like', 'Update' ) );
+
+		$args = array(
+			'posts_per_page' => $request->get_param( 'per_page' ),
+			'author'         => $user_id > 0 ? $user_id : null,
+			'paged'          => $page,
+			'post_type'      => Outbox::POST_TYPE,
+			'post_status'    => 'draft',
+
+			// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+			'meta_query'     => array(
+				array(
+					'key'     => '_activitypub_activity_type',
+					'value'   => $activity_types,
+					'compare' => 'IN',
+				),
+			),
+		);
+
+		if ( ! current_user_can( 'activitypub' ) ) {
+			$args['meta_query'][] = array(
+				'key'     => 'activitypub_content_visibility',
+				'value'   => array( ACTIVITYPUB_CONTENT_VISIBILITY_QUIET_PUBLIC, ACTIVITYPUB_CONTENT_VISIBILITY_LOCAL ),
+				'compare' => 'NOT IN',
+			);
```

Comment:
I think the current check requires the meta and for the public posts it is not, so they will not be included.

Shouldn't it be instead:

```suggestion
			$args['meta_query'][] = array(
				'key'     => 'activitypub_content_visibility',
				'compare' => 'NOT EXISTS',
			);
```